mk/build_optee_for_android.mk

```diff
@@ -0,0 +1,88 @@
+##########################################################
+## Common mk file used for Android to compile and       ##
+## integrate optee related components                   ##
+## Following variant need to be defined in device.mk    ##
+##    OPTEE_OS_DIR                                      ##
+##    OPTEE_TA_TARGETS                                  ##
+##    OPTEE_CFG_ARM64_CORE                              ##
+##    OPTEE_PLATFORM                                    ##
+## And  BUILD_OPTEE_MK need to be defined in device.mk  ##
+## to point to this file                                ##
+##                                                      ##
+## local_module need to be defined before include this  ##
+## this file to build TAs                               ##
+##                                                      ##
+##########################################################
+
+##########################################################
+## define common variants, like TA_DEV_KIT_DIR          ##
+##########################################################
+OPTEE_OUT_DIR ?= $(PRODUCT_OUT)/optee
+OPTEE_TA_OUT_DIR ?= $(OPTEE_OUT_DIR)/ta
+# Set so that OP-TEE clients can find the installed dev-kit, which
+# depends on platform and its optee word-size.
+OPTEE_OS_OUT_DIR ?= $(OPTEE_OUT_DIR)/arm-plat-$(OPTEE_PLATFORM)
+TA_DEV_KIT_DIR := $(OPTEE_OS_OUT_DIR)/export-${OPTEE_TA_TARGETS}
+
+TOP_ROOT_ABS := $(realpath $(TOP))
+CROSS_COMPILE64 := $(TOP_ROOT_ABS)/$(TARGET_TOOLS_PREFIX)
+CROSS_COMPILE_LINE := CROSS_COMPILE64="$(CROSS_COMPILE64)"
+ifeq ($(strip $($(combo_2nd_arch_prefix)TARGET_TOOLS_PREFIX)),)
+CROSS_COMPILE32 := $(TOP_ROOT_ABS)/$($(combo_2nd_arch_prefix)TARGET_TOOLS_PREFIX)
+CROSS_COMPILE_LINE += CROSS_COMPILE32="$(CROSS_COMPILE32)"
```

Comment:
No, only tested with Hikey, not tested with any 32bit build yet.
This is also my concerns, after all not verified on a real build.

If there is any problem need to be fixed for 32bit build, we only need to update the place here, so I think there is the value to have this mk file.
It will help to make the building and integrating OP-TEE components easier, and make the maintenance easier for Android build as well.